binding/binding.go

```diff
@@ -46,6 +46,10 @@ type BindingUri interface {
 	BindUri(map[string][]string, interface{}) error
 }
 
+type BindUnmarshaler interface {
+	UnmarshalParam(param []byte) error
```

Comment:
How about:
```suggestion
	UnmarshalParam(param string) error
```
That will save us some type conversions.

Reply:
I can do that. One reason perhaps not to do it is that there is already a [TextUnmarshaler](https://pkg.go.dev/encoding#TextUnmarshaler) interface in the go stdlib that uses `[]byte`.. though after reading your later comment about skipping it for structs, it might actually make sense to deviate. I'll put the reason why below